app/models/course_user.rb

```diff
@@ -176,6 +186,22 @@ def self.user?(user)
     all.exists?(user: user)
   end
 
+  # Soft delete the course user.
+  def destroy
+    ASSOCIATED_MODELS.each do |association|
+      send(association).destroy_all
+    end
+    super
+  end
+
+  # Restore the course user.
+  def restore
```

Comment:
When I try re-inviting user into a course, I get:
`PG::UniqueViolation: ERROR: duplicate key value violates unique constraint "index_course_users_on_course_id_and_user_id"
DETAIL: Key (course_id, user_id)=(course-xxx, user-yyy) already exists.`

Have you tested that users can be restored from the webapp?  

- instructor invites user into course using the Invite Users page
- user self-enrolls using course registration code (requires approval) from the Course Home page

Reply:
I have tested the scenario ``` Remove student from course and then enrol him again```.
`re-inviting user` The `invite` API has been updated.

Reply:
The API for "invitation" and "registration by code" has been updated. So all the scenarios to add user to course should be works now.